campaign-launcher/client/src/components/CampaignAddress/index.tsx

```diff
@@ -0,0 +1,90 @@
+import { FC, MouseEvent, useState } from "react"
+
+import { ChainId } from "@human-protocol/sdk"
+import ContentCopyIcon from '@mui/icons-material/ContentCopy';
+import { Box, IconButton, Tooltip, Typography } from "@mui/material"
+
+import { useIsMobile } from "../../hooks/useBreakpoints";
+import { OpenInNewIcon } from "../../icons"
+import { formatAddress, getExplorerUrl } from "../../utils"
+
+type Props = {
+  address: string;
+  chainId: ChainId;
+}
+
+const handleAddressClick = (
+  e: MouseEvent<HTMLButtonElement>,
+  chainId: ChainId,
+  address: string
+) => {
+  e.stopPropagation();
+  const explorerUrl = getExplorerUrl(chainId, address);
+  window.open(explorerUrl, '_blank');
+};
+
+const CampaignAddress: FC<Props> = ({ address, chainId }) => {
+  const [isCopied, setIsCopied] = useState(false);
+
+  const isMobile = useIsMobile();
+
+  const handleCopy = (e: MouseEvent<HTMLButtonElement>) => {
+    e.stopPropagation();
+    navigator.clipboard.writeText(address);
+    setIsCopied(true);
+    setTimeout(() => {
```

Comment:
Should we store timeout id in some ref and clean it up if component unmounts earlier than it fires?